recipes-kernel/linux/linux-yocto/0045-MCP2515-probe-using-ACPI-table-entry.patch

```diff
@@ -0,0 +1,109 @@
+From 6634df6b165436cbbd3ef265ad1f469d80d93504 Mon Sep 17 00:00:00 2001
+From: Sugnan Prabhu S <sugnan.prabhu.s@intel.com>
+Date: Wed, 20 Sep 2017 10:54:32 +0530
+Subject: [PATCH] MCP2515 probe using ACPI table entry
+
+Adds ACPI table to mcp251x driver and hard coding some of the values
+required for the driver initialization.
+
+Signed-off-by: Sugnan Prabhu S <sugnan.prabhu.s@intel.com>
+---
+ drivers/net/can/spi/mcp251x.c | 35 ++++++++++++++++++++++++++++++++---
+ 1 file changed, 32 insertions(+), 3 deletions(-)
+
+diff --git a/drivers/net/can/spi/mcp251x.c b/drivers/net/can/spi/mcp251x.c
+index 575790e..3f68400 100644
+--- a/drivers/net/can/spi/mcp251x.c
++++ b/drivers/net/can/spi/mcp251x.c
+@@ -75,6 +75,7 @@
+ #include <linux/slab.h>
+ #include <linux/spi/spi.h>
+ #include <linux/uaccess.h>
++#include <linux/acpi.h>
+ #include <linux/regulator/consumer.h>
+ 
+ /* SPI interface instruction set */
+@@ -219,6 +220,12 @@
+ 
+ #define DEVICE_NAME "mcp251x"
+ 
++#define CAN_SPI_BUS	1
++#define CAN_SPI_CS	0
++#define CAN_SPI_MZ	(1000000)
++#define OST_FREQ 25*1000*1000
++#define CPU_IN_MCP2515_IRQ 157
++
+ static int mcp251x_enable_dma; /* Enable SPI DMA. Default: 0 (Off) */
+ module_param(mcp251x_enable_dma, int, S_IRUGO);
+ MODULE_PARM_DESC(mcp251x_enable_dma, "Enable SPI DMA. Default: 0 (Off)");
+@@ -1033,12 +1040,22 @@ static int mcp251x_can_probe(struct spi_device *spi)
+ 	struct clk *clk;
+ 	int freq, ret;
+ 
++	spi->chip_select = CAN_SPI_CS;
++	spi->max_speed_hz = CAN_SPI_MZ;
++	spi->mode = SPI_MODE_0;
++
++	/* spi->irq = acpi_dev_gpio_irq_get(ACPI_COMPANION(&spi->dev), 0); */
++	spi->irq = CPU_IN_MCP2515_IRQ;
```

Comment:
What are the values coming from the ACPI table? Just override those that are wrong.
Explain in the commit message for the patch with the override the reason for each  override, including how the magic value of 157 was found.

Reply:
Also, surround the override with `if (has_acpi_companion()) { }` like is done everywhere else to maintain compatibility between ACPI and OF.